Uploader/Config.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Newtonsoft.Json;
+using NLog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+
+/// <summary>
+/// Data class that is deserialized from the JSON configuration file.
+/// </summary>
+public class Config
+{
+    private const string ConfigFileName = "Uploader.json";
+
+    /// <summary>
+    /// Path to the config file.
+    /// </summary>
+    public static readonly string ConfigFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, ConfigFileName);
+
+    /// <summary>
+    /// The Teamscale server to upload to.
+    /// </summary>
+    public TeamscaleServer Teamscale = null;
+
+    /// <summary>
+    /// The assembly from which to read the version number.
+    /// </summary>
+    [JsonProperty(Required = Required.Always)]
+    public string VersionAssembly;
+
+    /// <summary>
+    /// The directory to upload the traces to.
+    /// </summary>
+    public string Directory = null;
+
+    /// <summary>
+    /// Validates the configuration and returns all collected error messages. An empty list
+    /// means the configuration is valid.
+    /// </summary>
+    /// <returns></returns>
+    public List<string> Validate()
+    {
+        List<string> errorMessages = new List<string>();
```

Comment:
you'Re mixing validation with required json attribs. Imho I'd remove the JsonProperty stuff and do everything here...